packages/elements/src/components/file-picker/file-picker.component.ts

```diff
@@ -0,0 +1,301 @@
+          </svg>
+        `,
+        () => svg`
+          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 16 16" fill="currentColor" class="size-4">
+            <path fill-rule="evenodd" d="M2 4a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v8a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V4Zm2.22 1.97a.75.75 0 0 0 0 1.06l.97.97-.97.97a.75.75 0 1 0 1.06 1.06l1.5-1.5a.75.75 0 0 0 0-1.06l-1.5-1.5a.75.75 0 0 0-1.06 0ZM8.75 8.5a.75.75 0 0 0 0 1.5h2.5a.75.75 0 0 0 0-1.5h-2.5Z" clip-rule="evenodd" />
+          </svg>
+        `,
+      )}
+    `;
+  }
+
+  #handleFocus() {
+    this.shadowRoot?.querySelector('input')?.focus();
+  }
+
+  #prepareMap() {
+    if (this.rootModel == null) {
+      return;
+    }
+
+    const prepareFiles = <T extends FileUnderTestModel | TestFileModel>(
+      result: MetricsResult<T, Metrics | TestMetrics> | undefined,
+      parentPath: string | null = null,
+      allFilesKey: string,
+    ) => {
+      if (result === undefined) {
+        return;
+      }
+
+      if (result.file !== undefined && result.file !== null && result.name !== allFilesKey) {
+        this.#searchMap.set(parentPath == null ? result.name : `${parentPath}/${result.name}`, result.file);
+      }
+
+      if (result.childResults.length === 0) {
+        return;
+      }
+
+      result.childResults.forEach((child) => {
+        if (parentPath !== allFilesKey && parentPath !== null && result.name !== null) {
+          prepareFiles(child, `${parentPath}/${result.name}`, allFilesKey);
+        } else if ((parentPath === allFilesKey || parentPath === null) && result.name !== allFilesKey) {
+          prepareFiles(child, result.name, allFilesKey);
+        } else {
+          prepareFiles(child, null, allFilesKey);
+        }
+      });
+    };
+
+    prepareFiles(this.rootModel.systemUnderTestMetrics, null, 'All files');
+    prepareFiles(this.rootModel.testMetrics, null, 'All tests');
+  }
+
+  #handleKeyDown(event: KeyboardEvent) {
+    if ((event.ctrlKey || event.metaKey) && event.key === 'k') {
+      this.#togglePicker(event);
+    } else if (!this.openPicker && event.key === '/') {
+      this.#togglePicker(event);
+    } else if (event.key === 'Escape') {
+      this.#closePicker();
+    } else if (event.key === 'ArrowUp') {
+      this.#handleArrowUp();
+    } else if (event.key === 'ArrowDown') {
+      this.#handleArrowDown();
+    }
+
+    if (event.key === 'ArrowUp' || event.key === 'ArrowDown') {
+      setTimeout(() => {
+        this.#scrollActiveLinkInView();
+      });
+    }
+
+    if (event.key === 'Enter') {
+      this.#handleEnter();
+    }
+  }
+
+  #scrollActiveLinkInView() {
+    const activeLink = this.shadowRoot?.querySelector('a[data-active]');
+
+    if (activeLink != null) {
```

Comment:
```suggestion
    if (activeLink !== null) {
```

Or: `activeLink?.scrollIntoView`